core/src/Kokkos_View.hpp

```diff
@@ -1422,8 +1429,12 @@ class View : public ViewTraits<DataType, Properties...> {
   KOKKOS_DEFAULTED_FUNCTION
   View() = default;
 
-  KOKKOS_DEFAULTED_FUNCTION
-  View(const View&) = default;
+  KOKKOS_INLINE_FUNCTION
+  View(const View& rhs) : m_map(rhs.m_map), m_track(*this, rhs) {}
```

Comment:
I am a bit concerned that the default copy constructible is not possible anymore. Need justification. (i.e. why we need to make the standard copy construction something else than "default").

Reply:
Also, passing a not-fully constructed instance of `View` to `m_track` is unwise at best (and undefined behavior at worst)

Reply:
This makes me very uncomfortable, first I'm not convinced this isn't UB, second this means the m_track constructor actually needs to know in its constructor that the argument passed in is only partially constructed! It seems terrible to me if 
`void foo(auto &thing){thing.x;}` That x may not be constructed yet.

Reply:
@nliber has told me that this isn't UB, iff the constructor is  careful about how it uses the `View`, but I still think it's a bad idea.

Reply:
so, the constructor is still not default, but we no longer use *this 60b42a7d87599a9d2c2bfe913cb811b38a3b6f9f